src/lib/cimb_translator/CimbReader.cpp

```diff
@@ -141,6 +148,114 @@ bool CimbReader::done() const
 	return !_good or _positions.done();
 }
 
+void CimbReader::init_ccm(unsigned color_bits, unsigned interleave_blocks, unsigned interleave_partitions, unsigned fountain_blocks)
+{
+	if (_colorCorrection != 2)
+		return;
+
+	// if no fountain header, we don't attempt color correction
```

Comment:
This method coordinates the new color correction logic. It needs a refactor, but the comments hopefully explain the basics (cimbar-py also contains an implementation).